assembly/src/parsers/mod.rs

```diff
@@ -205,19 +205,12 @@ pub fn parse_program(source: &str) -> Result<ProgramAst, ParsingError> {
 
     // parse the sequence of nodes and add each node to the list
     let mut end_of_nodes = false;
-    let beginning_node_count = body.len();
     while !end_of_nodes {
         let node_count = body.len();
         context.parse_body(&mut tokens, &mut body, false)?;
         end_of_nodes = body.len() == node_count;
     }
```

Comment:
Looking at this code now, I'm not sure we need this `while` loop here (and I also think that the comment for `ParserContext::parse_body()` function might be wrong).

It is not really a part of this PR - so, if we don't fix it here, might be good to create an issue for this.